karapace/backup/api.py

```diff
@@ -461,10 +458,16 @@ def restore_backup(
             elif isinstance(instruction, ProducerSend):
                 if producer is None:
                     raise RuntimeError("Backend has not yet sent RestoreTopic.")
-                _handle_producer_send(instruction, producer)
+                _handle_producer_send(instruction, producer, _producer_error_callback)
+                # Immediately check if producer send generated an exception
```

Comment:
issue: Please, let's document the _why_ here, not only the _what_. The comments still don't communicate that this call is an optimization that in some cases might reduce time-to-error, and they don't document that the other call is _essential_ for correct behavior.